### ISSUE TITLE
Shit error log as part of Dashboard 

### ISSUE THREAD

**Issue by matentzn** [OBOFoundry/OBO-Dashboard#34]
This way we have a chance to see why certain things did not work. (Like recently some ontology parses)

**Comment by matentzn** [comment]
Awesome typo in issue title. Sorry :D :D

**Comment by dosumis** [comment]
+1 - I expected to get full report from TSV download link, e.g. here http://dashboard.obofoundry.org/dashboard/fbbt/robot_report.html - but instead it clips at 200.

**Comment by matentzn** [comment]
The problem with that is that for some ontologies, like ncbitaxon, the log is so large it is outside the file limit. Also, we have an issue with diffs in the report which badly pollute the git history when committing large reports. Would you be happy with a compromise of increasing the cap at 1000?

**Comment by jamesaoverton** [comment]
I agree with @matentzn that we don't want to be committing huge log files to this repository.

One option we've discussed is pushing the big logs to some other storage such as S3, or some variation on Git Large File Storage.

It just occurred to me as I replied to #35 that if we filter the ROBOT reports somehow we could also be a bit more clever and sample a few messages of each type. So the reports would still be small but have more variety and maybe be more useful.

These could both be pursued.

**Comment by matentzn** [comment]
Yeah that is a good idea. We could:
- Include all ERROR (these are typically few)
- Increase the number of rows to 1000
- Strip out owlclassexpression hashcodes to reduce the diff
- stratify the cut-off by shrinking large groups where necessary

I think the last will only impact very few ontologies.

**Comment by dosumis** [comment]
When trimming, it would be useful to have a few examples of each type of error  - as once you've seen a few examples you can often see a pattern to them.  In this example:

Rule | Number of errors
-- | --
duplicate_definition | 462
duplicate_exact_synonym | 953
missing_definition | 426
missing_obsolete_label | 626
lowercase_definition | 308
missing_superclass | 3022

If limited to 1000 and taken in this order, we wouldn't see any examples after the second one.

**Comment by matentzn** [comment]
Sorry I realise I expressed it badly - that's exactly what I meant by "stratify the cut-off by shrinking large groups where necessary" - and what @jamesaoverton suggests. This is what we will do.

**Comment by dosumis** [comment]
(I should read previous more carefully!)

**Comment by anitacaron** [comment]
To clarify, the original issue here is related to the OBO Dashboard running process. When we run the dashboard, we cannot easily check the errors raised in the process because the process doesn't stop. The easy solution would be to output the log to a file but not commit it. Before pushing the new dashboard release, we could check the errors and probably rerun the dashboard.
